### PR TITLE
Enable DDP Plugin to pass through args to LightningDistributedDataParallel

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

This makes the DDP Plugin more generic, so people don't have to subclass if they only want to change what goes into the `DistributedDataParallel` constructor. This preserves `find_unused_parameters=True` as the default.

Related: is `find_unused_parameters` forced to be True since Lightning wraps the whole LightningModule in `LightningDistributedDataParallel` ?

cc @williamFalcon 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/ddp_plugin.py [hunk 1]
```diff
@@ -17,11 +21,16 @@ def configure_ddp(self, model, device_ids):
 
         my_ddp = MyDDP()
         trainer = Trainer(accelerator='ddp_x', plugins=[my_ddp])
-
     """
 
-    def configure_ddp(self, model: LightningModule, device_ids: List[int]) -> LightningDistributedDataParallel:
+    def __init__(self, **kwargs):
+        self._ddp_args: Dict[str, Any] = kwargs
```
Comment (awaelchli) [inline pytorch_lightning/plugins/ddp_plugin.py:27]: let's call it ddp_kwargs?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/ddp_plugin.py (hunk 1 above)
Comment (ananthsub) [inline pytorch_lightning/plugins/ddp_plugin.py:27]:
```suggestion
        self.ddp_kwargs: Dict[str, Any] = kwargs
```

File: pytorch_lightning/plugins/ddp_plugin.py [hunk 2]
```diff
@@ -43,5 +52,13 @@ def configure_ddp(self, model, device_ids):
...
-        model = LightningDistributedDataParallel(model, device_ids=device_ids, find_unused_parameters=True)
+        # if unset, default `find_unused_parameters` `True`
+        self._ddp_args["find_unused_parameters"] = self._ddp_args.get(
+            "find_unused_parameters", True
+        )
+        model = LightningDistributedDataParallel(
+            model,
+            device_ids=device_ids,
+            **self._ddp_args,
+        )
```
Comment (ananthsub) [inline pytorch_lightning/plugins/ddp_plugin.py:63]:
```suggestion
        self.ddp_kwargs["find_unused_parameters"] = self.ddp_kwargs.get(
            "find_unused_parameters", True
        )
        model = LightningDistributedDataParallel(
            model,
            device_ids=device_ids,
            **self.ddp_kwargs,
        )
```

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/ddp_plugin.py [hunk 3]
```diff
@@ -17,11 +21,16 @@ def configure_ddp(self, model, device_ids):
 
         my_ddp = MyDDP()
         trainer = Trainer(accelerator='ddp_x', plugins=[my_ddp])
-
     """
 
-    def configure_ddp(self, model: LightningModule, device_ids: List[int]) -> LightningDistributedDataParallel:
+    def __init__(self, **kwargs):
+        self.ddp_kwargs: Dict[str, Any] = kwargs
```
Comment (tchaton) [inline pytorch_lightning/plugins/ddp_plugin.py:27]: Would you mind making this one private `self._ddp_kwargs `

**Review by tchaton — APPROVED** [review]
Awesome addition !

File: pytorch_lightning/plugins/ddp_plugin.py [hunk 4]
```diff
@@ -1,12 +1,16 @@
-from pytorch_lightning.overrides.data_parallel import LightningDistributedDataParallel
-from pytorch_lightning.core.lightning import LightningModule
 from typing import List
```
Comment (SkafteNicki) [inline pytorch_lightning/plugins/ddp_plugin.py:1]:
```suggestion
from typing import List, Dict, Any
```

**Pull request merged**